### PR TITLE
SAMZA-469: Update Scala version to 2.11

### PR DESCRIPTION
**Pull request by craigtmc**

### CODE REVIEW

File: bin/check-all.sh [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
 set -e
 
-SCALAs=( "2.10" )
+SCALAs=( "2.10", "2.11" )
```
Comment (nickpan47) [inline bin/check-all.sh]: The list should be separated by white spaces, not ","

File: bin/check-all.sh [hunk 2]
```diff
@@ -21,7 +21,7 @@
...
 JDKs=( "JAVA7_HOME" "JAVA8_HOME" )
```
Comment (nickpan47) [inline bin/check-all.sh]: I realized that JAVA7_HOME should be removed here. Creating another ticket for this.

File: bin/check-all.sh [hunk 3]
```diff
@@ -21,7 +21,7 @@
 
 set -e
 
-SCALAs=( "2.10", "2.11" )
+SCALAs=( "2.10" "2.11" )
```
Comment (nickpan47) [inline bin/check-all.sh]: Thanks! Looks good to me.

File: gradle.properties [hunk 4]
```diff
@@ -16,7 +16,7 @@
 # under the License.
 group=org.apache.samza
 version=0.11.1-SNAPSHOT
-scalaVersion=2.10
+scalaVersion=2.11
```
Comment (nickpan47) [inline gradle.properties]: @craigtmc LinkedIn is still using Scala 2.10 internally. We are validating whether this change would work w/ our internal build (w/ default to 2.11). Once that is validated, we can go ahead w/ this PR. Thanks!

**Comment by shanthoosh** [comment]
We've tested and validated the usage of Scala 2.11 with the internal build at LinkedIn. This looks good.

**Pull request closed without merging**